### DISCUSSION TITLE
running tatool apps in Qualtrics surveys

### DISCUSSION THREAD

**Discussion by dickinsonca** [tatool/tatool-web#165] [Q&A] [answered]
I'm wondering if it is possible to run a tatool app within a Qualtrics survey - I'm trying to develop a complex working-memory span task based on the automated OSPAN and RSPAN tasks developed by Unsworth et al. (2005).  Any information would be appreciated - thank you in advance.

**Comment by cvonbastian** [comment, marked as answer]
Hi there! You can redirect from Qualtrics to Tatool Web and the reverse (using "redirect URL" in the module editor), see also [here](https://github.com/tatool/tatool-web/discussions/148). Is that what you've been looking for?

**Reply by dickinsonca** [reply]
Yes it is.  I would also need to pass an ID code (that would be embedded data in the Qualtrics survey) to Tatool so it could be embedded in the user's data file.  Do you know if that would be possible?  Any information would be greatly appreciated (at this point, I'm working on developing the application in PsychoPy, which converts the Python code to Javascript, but I think I'd be better off developing the application in a Javascript-based environment).  Thank you.

**Reply by alocher** [reply]
You can pass an identifer From Qualtrics to Tatool by passing the extid parameter as explained here under ["Publish"](https://www.tatool-web.com/#!/doc/use-editor.html).

The other way around is also possible as Tatool passes information to Qualtrics which can be used with embedded data in Qualtrics.
See answer [here](https://github.com/tatool/tatool-web/discussions/151#discussioncomment-485292).
